rundeckapp/test/unit/rundeck/quartzjobs/ExecutionJobSpec.groovy

```diff
@@ -169,4 +169,62 @@ class ExecutionJobSpec extends Specification {
 
 
     }
+    def "scheduled job was stoped by another cluster node, so should be deleted from quartz scheduler"() {
+        given:
+        def serverUUID = UUID.randomUUID().toString()
+        def jobUUID = UUID.randomUUID().toString()
+        def es = Mock(ExecutionService)
+        def eus = Mock(ExecutionUtilService)
+        def fs = Mock(FrameworkService) {
+            getServerUUID() >> serverUUID
+        }
+        ScheduledExecution se = new ScheduledExecution(
+                jobName: 'blue',
+                project: 'AProject',
+                groupPath: 'some/where',
+                description: 'a job',
+                argString: '-a b -c d',
+                serverNodeUUID: serverUUID,
+                workflow: new Workflow(
+                        keepgoing: true,
+                        commands: [new CommandExec(
+                                [adhocRemoteString: 'test buddy', argString: '-delay 12 -monkey cheese -particle']
+                        )]
+                ),
+                scheduled: false,
+                executionEnabled: true,
+                scheduleEnabled: false
+        )
+        se.save(flush:true)
+        def datamap = new JobDataMap(
+                [
+                        scheduledExecutionId: se.id,
+                        executionService    : es,
+                        executionUtilService: eus,
+                        frameworkService    : fs,
+                        bySchedule          : true,
+                        serverUUID          : serverUUID
+                ]
+        )
+        ExecutionJob job = new ExecutionJob()
+        def ajobKey = JobKey.jobKey('jobname', 'jobgroup')
+
+        def quartzScheduler = Mock(Scheduler)
+        def context = Mock(JobExecutionContext) {
+            getJobDetail() >> Mock(JobDetail) {
+                getJobDataMap() >> datamap
+                getKey() >> ajobKey
+            }
+
+            getScheduler() >> quartzScheduler
+        }
+
+        when:
+        job.execute(context)
+
+        then:
+        1 * quartzScheduler.deleteJob(ajobKey)
+
```

Comment:
Here you could use:

~~~ {.groovy}
where:
isscheduled | isexecenabled | isscheduleenabled
false | true | true
true | false | true
true | true | false
~~~